packages/opentelemetry-core/src/common/id-util.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as crypto from 'crypto';
+
+const SPAN_ID_BYTES = 8;
+
+/**
+ * Returns a random 16-byte trace ID formatted/encoded as a 32 lowercase hex
+ * characters corresponding to 128 bits.
+ */
+export function randomTraceId(): string {
+  return randomSpanId() + randomSpanId();
+}
+
+/**
+ * Returns a random 8-byte span ID formatted/encoded as a 16 lowercase hex
+ * characters corresponding to 64 bits.
+ */
+export function randomSpanId(): string {
+  return crypto.randomBytes(SPAN_ID_BYTES).toString('hex');
```

Comment:
This is a very heavy operation. We used this approach in the past but switched to a pseudo-random generator instead with `crypto` only used to generate a random seed for the process.

Reply:
Since this will make the code a bit more involved, maybe `crypto` can be used for now and let's create an issue to make it faster?

Reply:
Calling crypto twice imposes much overhead, compared to an approach like I did in OpenCensus: https://github.com/census-instrumentation/opencensus-node/pull/198

Basically there is a function that can create both trace-id and span-id by just calling crypto once.
The PR also contains some benchmarks.

Reply:
This is still too slow. There should only be one call to `crypto` for the entire process to generate a seed and then it becomes possible to generate IDs using a pseudo-random algorithm. This would be several thousands of percents faster.

A safer alternative could be to generate a cryptographically secure trace ID and a pseudo-random span ID with the trace ID as seed, but I don't think that's warranted (we haven't hit a collision in years using pseudo-random numbers).